pkg/hpsf/exporters.go

```diff
@@ -0,0 +1,227 @@
+package hpsf
+
+// ExporterType represents the different types of exporters available in HPSF
+type ExporterType string
+
+const (
+	ExporterTypeHoneycomb         ExporterType = "Honeycomb"
+	ExporterTypeAWSS3             ExporterType = "AWSS3"
+	ExporterTypeEnhanceIndexingS3 ExporterType = "EnhanceIndexingS3"
+	ExporterTypeOTelGRPC          ExporterType = "OTelGRPC"
+	ExporterTypeOTelHTTP          ExporterType = "OTelHTTP"
+	ExporterTypeDebug             ExporterType = "Debug"
+	ExporterTypeNop               ExporterType = "Nop"
+)
+
+// ExporterInfo represents an exporter component with its type and metadata
+type ExporterInfo struct {
+	// Type is the exporter type (e.g., "Honeycomb", "AWSS3")
+	Type ExporterType
+	// Metadata contains exporter-specific configuration details
+	Metadata ExporterMetadata
+}
+
+// ExporterMetadata is an interface for exporter-specific metadata
+type ExporterMetadata interface {
+	// GetType returns the exporter type
+	GetType() ExporterType
+}
```

Comment:
Since `ExporterInfo` holds the `ExporterType` we shouldn't need `ExporterMetadata` to also return the `ExporterType`